readthedocs_build/config/config.py

```diff
@@ -205,6 +222,50 @@ def validate_base(self):
             base = validate_directory(base, base_path)
         self['base'] = base
 
+    def validate_build(self):
+        """
+        Validate the build config settings.
+
+        This is a bit complex,
+        so here is the logic:
+
+        * We take the default image & version if it's specific in the environment
+        * Then update the _version_ from the users config
+        * Then append the default _image_, since users can't change this
+        * Then update the env_config with the settings for that specific image
+           - This is currently used for a build image -> python version mapping
+
+        This means we can use custom docker _images_,
+        but can't change the supported _versions_ that users have defined.
+        """
+        # Defaults
+        if 'build' in self.env_config:
+            build = self.env_config['build']
+        else:
+            build = {'image': DOCKER_IMAGE}
+
+        # User specified
+        if 'build' in self.raw_config:
+            _build = self.raw_config['build']
+            if 'image' in _build:
+                with self.catch_validation_error('build'):
+                    build['image'] = validate_choice(
+                        str(_build['image']),
+                        self.DOCKER_SUPPORTED_VERSIONS,
+                    )
+            if ':' not in build['image']:
```

Comment:
If the build image is passed from RTD it will be fully qualified already.

Reply:
Also the default (not from YAML)

Reply:
Oh I see, it's only checking for when build is in config :D

Reply:
Not sure if I follow you. If the image is passed from RTD or not from YAML we won't be inside `if 'build' in self.raw_config` since I understood that `raw_config` is the YAML. So, maybe this if should be outside this block.